### PR TITLE
Port project to Elm 0.19

### PR DESCRIPTION
**Pull request by knuton**
Upgraded using `elm-upgrade`, manual fixing of tests and especially adapting to changed API of `elm/parser` was necessary.

```
INFO: Found elm at /nix/store/iqcz6gg0g04dshgblrqdasqdiqljmc3c-elm-0.19.0/bin/elm
INFO: Found elm 0.19.0
INFO: Found elm-format at /nix/store/9520s4y36qarzklbz90dv1qbbq1m70j3-elm-format-0.8.1/bin/elm-format
INFO: Found elm-format 0.8.1
INFO: Cleaning ./elm-stuff before upgrading
INFO: Converting elm-package.json -> elm.json
INFO: Detected a package project (this project has exposed modules)
INFO: Switching from elm-lang/core (deprecated) to elm/core
INFO: Installing latest version of elm/core
INFO: Detected use of elm-lang/core#Json.Decode; installing elm/json
INFO: Switching from elm-tools/parser (deprecated) to elm/parser
INFO: Installing latest version of elm/parser
INFO: Upgrading *.elm files in src//


SUCCESS! Your project's dependencies and code have been upgraded.
However, your project may not yet compile due to API changes in your
dependencies.

See <https://github.com/elm/compiler/blob/master/upgrade-docs/0.19.md>
and the documentation for your dependencies for more information.

Here are some common upgrade steps that you will need to do manually:

- elm/core
  - [ ] Replace uses of toString with String.fromInt, String.fromFloat, or Debug.toString as appropriate
```

### CODE REVIEW

**Comment by knuton** [comment]
@guyonvarch You might be happy about the new parser for `alphanumericIdentifier`. :wink:

File: src/Semver.elm [hunk 1]
```diff
@@ -253,53 +260,57 @@ buildMetadataIdentifier =
...
+    Parser.loop
+        [ firstItem ]
+        (\segments ->
+            Parser.oneOf
+                [ Parser.succeed (\item -> Parser.Loop (item :: segments))
+                    |. Parser.chompIf ((==) '.')
+                    |= identifier
+                , Parser.succeed (Parser.Done (List.reverse segments))
+                ]
+        )
```
Comment (guyonvarch) [inline src/Semver.elm:279]: Alternative solution using `Parser.sequence`:

```elm
identifierSeries : Parser i -> Parser a -> Parser (List i)
identifierSeries identifier separator =
    Parser.succeed identity
        |. separator
        |= sequence1 identifier (Parser.chompIf ((==) '.'))


sequence1 : Parser e -> Parser s -> Parser (List i)
sequence1 elem sep =
    Parser.succeed (::)
        |= elem
        |= Parser.sequence
            { start = ""
            , separator = ""
            , end = ""
            , spaces = Parser.succeed ()
            , item = Parser.succeed identity |. sep |= elem
            , trailing = Parser.Forbidden
            }
```

File: src/Semver.elm [hunk 2]
```diff
@@ -253,53 +260,57 @@ buildMetadataIdentifier =
...
-                (++)
-                (Parser.keep oneOrMore isDigit)
-                startsWithNonDigit
-            ]
+    Parser.getChompedString <|
+        -- Delay commit until after something other than a digit has been seen,
+        -- this makes the composed parser usable as a oneOf option.
+        Parser.backtrackable (Parser.chompWhile isDigit)
+            |. Parser.chompIf isNonDigit
+            |. Parser.chompWhile isIdentifierCharacter
```
Comment (guyonvarch) [inline src/Semver.elm:291]: Nice!

**Review by guyonvarch — APPROVED** [review]

File: src/Semver.elm (hunk 1 above)
Comment (knuton) [reply src/Semver.elm:279]: It works, but I think following [the note in the `elm/parser` docs](https://package.elm-lang.org/packages/elm/parser/1.1.0/Parser#sequence) makes sense:

> It is better to get nice error messages with a lower-level implementation than to try to hack high-level parsers to do things they are not made for.

Not much hacking here yet, but a lot of things that `sequence` expects have to be "nullified".

**Pull request merged**